modules/howtos/pages/kv-operations.adoc

```diff
@@ -22,33 +22,100 @@ Or read on, for a hands-on introduction to working with documents from the .NET
 The core interface to Couchbase Server is simple KV operations on full documents.
 Make sure you're familiar with the basics of authorization and connecting to a Cluster from the xref:hello-world:start-using-sdk.adoc[Start Using the SDK section].
 
-See the https://github.com/couchbase/docs-sdk-c/blob/release/3.1/modules/devguide/examples/c/retrieving.cc[code sample^] for use in context.
+See the https://github.com/couchbase/docs-sdk-c/blob/release/3.0/modules/devguide/examples/c/retrieving.cc[code sample^] for use in context.
+
+== Upsert
+
+[TIP]
+.Sub-Document Operations
+====
+All of these operations involve fetching the complete document from the Cluster.
+Where the number of operations or other circumstances make bandwidth a significant issue, the SDK can work on just a specific _path_ of the document with xref:subdocument-operations.adoc[Sub-Document Operations].
+====
+
 ////
-We're going to expand on the short _Upsert_ example we used there, adding options as we move through the various CRUD operations.
-Here is the _Insert_ operation at its simplest:
+== Retrieving full documents
+
+Using the `Get()` method with the document key can be done in a similar fashion to the other operations:
 
 [source,csharp]
 ----
-Insert
-var document = new {foo = "bar", bar = "foo"};
-var result = await collection.Insert("document-key", document);
+var result = await collection.Get("document-key");
```

Comment:
This might have slipped in through, but looks like this is C#, might need to update this and the examples below.

Reply:
you right, this is c# snippet